packages/runtime/src/index.ts

```diff
@@ -136,7 +136,12 @@ const plugin: NetlifyPlugin = {
 
         await updateRequiredServerFiles(publish, config)
       } else {
-        const nextAuthUrl = `${process.env.DEPLOY_PRIME_URL}${basePath}`
+        // Using the deploy prime url in production leads to issues because the unique deploy ID is part of the generated URL
+        // and will not match the expected URL in the callback URL of an OAuth application.
+        const nextAuthUrl =
+          process.env.CONTEXT === 'production'
+            ? `${process.env.URL}${basePath}`
+            : `${process.env.DEPLOY_PRIME_URL}${basePath}`
```

Comment:
[dust] Tiny refactor suggestion.

```suggestion
        const nextAuthUrl = `${
          process.env.CONTEXT === 'production' ? process.env.URL : process.env.DEPLOY_PRIME_URL
        }${basePath}`
```